### DISCUSSION TITLE
Transform response before caching

### DISCUSSION THREAD

**Discussion by amrendr-cb** [arthurfiorette/axios-cache-interceptor#771] [Q&A] [answered]
There are times when I don't want to cache the full response as its too big (causing memory pressure) and I only need few data point from response for my further use.

is it possible to transform my response data before it is saved in the cache and return the transformed data as part of axios response. I tried using axios response interceptor but all other axios response (non cached) also intercepted and it difficult to keep separation between the different calls.

can I defined some data mapper while creating a cached axios instance?

**Comment by arthurfiorette** [comment]
Hey @amrendr-cb, currently this is not possible by default, you may make this work by applying your own interceptor that maps the response before the axios-cache-interceptor's interceptor run.

However, this would be an awesome feature, are you open to a PR?

**Reply by amrendr-cb** [reply]
@arthurfiorette, I have opened a PR https://github.com/arthurfiorette/axios-cache-interceptor/pull/775

**Reply by amrendr-cb** [reply, marked as answer]
We can use the `transformResponse` API of Axios to provide our transformation method. After data transformation, the cache interceptor stores the transformed data in the cache. 

With the fix #775, the cache interceptor no longer runs `transformResponse` twice on cached requests, which was causing unintended responses.
